packages/ui/src/components/LooMap.js

```diff
@@ -182,6 +210,114 @@ export class LooMap extends Component {
     }
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    // New cluster layer, we probably remounted
+    if (prevState.clusterLayer !== this.state.clusterLayer) {
+      // We need to re-add everything
+      this.addRemoveMarkers(_.keyBy(this.props.loos, '_id'), {});
+      return;
+    }
+
+    // Otherwise, we only care if markers and their icons could've changed
+    if (
+      prevProps.loos === this.props.loos &&
+      prevProps.highlight === this.props.highlight &&
+      prevProps.countLimit === this.props.countLimit &&
+      prevProps.countFrom === this.props.countFrom
+    ) {
+      return;
+    }
+
+    let loosThen = _.keyBy(prevProps.loos, '_id');
+    let loosNow = _.keyBy(this.props.loos, '_id');
+
+    // Remove elements that have unchanged location from both, effectively
+    // diff'ing; this will leave us with markers to remove in loosThen and
+    // markers to add in loosNow
+    for (let id of Object.keys(loosNow)) {
+      if (
+        loosThen.hasOwnProperty(id) &&
+        _.isEqual(loosThen[id].geometry, loosNow[id].geometry)
```

Comment:
We might gain a shred of simplicity/performance here by comparing the geohashes? But wait... Isn't the _id enough?

Reply:
There are two potential causes of the `loos` prop updating:

1. The set of nearby loos has changed (e.g. if the user has moved) _and/or_
1. A loo's properties have been changed (e.g. after the user has just edited it)

Checking just `_id` is enough to ensure that we keep the view up-to-date for the first case but we must compare location to keep the view up-to-date for the second.

While it's not ideal to remove and re-add a loo to handle it moving, this should rarely be done more than once per call to the function so it should be fine for now.

On the other hand, we are checking equality of the geometry object for as many loos as there are within 50km on each function call, so it's probably worth moving to `geohash` there; I'll get a commit up for that.